packages/wallet-sdk/src/sign/scw/utils/createSubAccountSigner.ts

```diff
@@ -0,0 +1,90 @@
+import { Hex, http, SignableMessage, TypedDataDefinition } from 'viem';
+import { createPaymasterClient, toCoinbaseSmartAccount } from 'viem/account-abstraction';
+import { baseSepolia } from 'viem/chains';
+
+import { RequestArguments } from ':core/provider/interface.js';
+import { getBundlerClient, getClient } from ':stores/chain-clients/utils.js';
+import { SubAccount, SubAccountInfo } from ':stores/sub-accounts/store.js';
+
+export async function createSubAccountSigner(subaccount: SubAccountInfo) {
+  const { getSigner } = SubAccount.getState();
+  if (!getSigner) {
+    throw new Error('get signer not found');
+  }
+
+  const signer = await getSigner();
+  if (!signer) {
+    throw new Error('signer not found');
+  }
+
+  // TODO[jake] how do we handle unsupported chains
+  const client = getClient(subaccount.chainId ?? baseSepolia.id);
+  if (!client) {
+    throw new Error('client not found');
+  }
+
+  const account = await toCoinbaseSmartAccount({
+    address: subaccount.address,
+    client,
+    owners: [subaccount.root, signer],
+    ownerIndex: 1,
+  });
+
+  return {
+    request: async (args: RequestArguments): Promise<Hex> => {
+      switch (args.method) {
+        case 'eth_sendTransaction': {
+          return account.sign(args.params as { hash: Hex });
+        }
+        case 'wallet_sendCalls': {
+          if (!Array.isArray(args.params)) {
+            throw new Error('invalid params');
```

Comment:
think we should throw standardErrors.rpc.invalidParams error in this file, we might wanna want to only throw standardError in SDK

Reply:
good callout let me update